src/test/java/com/example/ticketbooking/service/RoomServiceTest.java

```diff
@@ -0,0 +1,61 @@
+package com.example.ticketbooking.service;
+
+import com.example.ticketbooking.exception.IncorrectReservationException;
+import com.example.ticketbooking.exception.LimitReachedException;
+import com.example.ticketbooking.model.Room;
+import com.example.ticketbooking.repository.RoomRepository;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+
+@SpringBootTest
+class RoomServiceTest {
+
+    @Autowired
+    private RoomService cut;
+
+    @MockBean
+    private RoomRepository repo;
+
+
+
+    @Test
+    void whenAddRoom_thenRoomIsSaved() throws IncorrectReservationException, LimitReachedException {
+        int rows = 7;
+        int seatsPerRow = 10;
+
+        Mockito.when(repo.findAll()).thenReturn(new ArrayList<>());
+
+        cut.create(rows, seatsPerRow);
+
+        Mockito.verify(repo).save(any());
```

Comment:
Jak dla mnie to ten test sprawdza tylko czy została zawołana metoda save, ale co zostało do niej przekazane to już nie wiadomo... widzę kilka scenariuszy jak zepsuć kod klasy `RoomService` tak żeby test dalej był zielony :)

Reply:
to prawda